nemo/collections/asr/modules/audio_modules.py

```diff
@@ -0,0 +1,686 @@
+
+            raise ModuleNotFoundError(
+                "torchaudio is not installed but is necessary to instantiate a {self.__class__.__name__}"
+            )
+
+        super().__init__()
+
+        # For now, assume FFT length is divisible by two
+        if fft_length % 2 != 0:
+            raise ValueError(f'fft_length = {fft_length} must be divisible by 2')
+
+        self.istft = torchaudio.transforms.InverseSpectrogram(
+            n_fft=fft_length, hop_length=hop_length, pad_mode='constant'
+        )
+
+        self.F = fft_length // 2 + 1
+
+    @property
+    def num_subbands(self) -> int:
+        return self.F
+
+    @property
+    def input_types(self) -> Dict[str, NeuralType]:
+        """Returns definitions of module output ports.
+        """
+        return {
+            "input": NeuralType(('B', 'C', 'D', 'T'), SpectrogramType()),
+            "input_length": NeuralType(('B',), LengthsType()),
+        }
+
+    @property
+    def output_types(self) -> Dict[str, NeuralType]:
+        """Returns definitions of module output ports.
+        """
+        return {
+            "output": NeuralType(('B', 'C', 'T'), AudioSignal()),
+            "output_length": NeuralType(('B',), LengthsType()),
+        }
+
+    @typecheck()
+    def forward(self, input: torch.Tensor, input_length: torch.Tensor) -> torch.Tensor:
+        """Convert input complex-valued spectrogram to a time-domain
+        signal. Multi-channel IO is supported.
+
+        Args:
+            input: Input spectrogram for C channels, shape (B, C, F, N)
+            input_length: Length of valid entries along the time dimension, shape (B,)
+
+        Returns:
+            Time-domain signal with T time-domain samples and C channels, (B, C, T)
+            and output length with shape (B,).
+        """
+        output_length = self.get_output_length(input_length=input_length)
+
+        B, F, N = input.size(0), input.size(-2), input.size(-1)
+        assert F == self.F, f'Number of subbands F={F} not matching self.F={self.F}'
+        input = input.view(B, -1, F, N)
+
+        # iSTFT output (B, C, T)
+        with torch.cuda.amp.autocast(enabled=False):
+            output = self.istft(input.cdouble())
+
+        # Mask padded samples
+        length_mask: torch.Tensor = make_seq_mask_like(
+            lengths=output_length, like=output, time_dim=-1, valid_ones=False
+        )
+        output = output.masked_fill(length_mask, 0.0)
+
+        return output, output_length
+
+    def get_output_length(self, input_length: torch.Tensor) -> torch.Tensor:
+        """Get length of valid samples for the output.
+
+        Args:
+            input_length: number of valid frames, shape (B,)
+
+        Returns:
+            Number of valid samples, shape (B,)
+        """
+        output_length = input_length.sub(1).mul(self.istft.hop_length).long()
+        return output_length
+
+
+class MultichannelFeatures(NeuralModule):
```

Comment:
Same as above

Reply:
Feels like variable name. may be add `processor` at the end.